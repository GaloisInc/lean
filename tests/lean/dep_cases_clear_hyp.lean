universes u v

inductive tree (α : Type u)
| leaf {} : tree
| node (l : tree) (v : α) (r : tree) : tree

namespace tree
variables {α : Type u}

inductive is_searchable (lt : α → α → Prop) : tree α → α → α → Prop
| leaf_s  {lo hi}       : lt lo hi → is_searchable leaf lo hi
| node_s  {l r v lo hi} (hs₁ : is_searchable l lo v) (hs₂ : is_searchable r v hi) : is_searchable (node l v r) lo hi

example (l r : tree α) {lo v hi lt} (h : is_searchable lt (node l v r) lo hi) : true :=
begin
  cases h,
  trace_state, /- Should not contain h -/
  trivial
end

end tree
