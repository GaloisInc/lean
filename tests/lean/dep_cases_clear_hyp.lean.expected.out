α : Type u,
l r : tree α,
lo v hi : α,
lt : α → α → Prop,
h_hs₁ : is_searchable lt l lo v,
h_hs₂ : is_searchable lt r v hi
⊢ true
